core/tests/ledger_cleanup.rs

```diff
@@ -510,16 +554,30 @@ mod tests {
                 &sys.get_stats(),
             );
 
-            if stop_size_bytes > 0 {
-                if storage_previous >= stop_size_bytes {
-                    stop_size_bytes_exceeded_iterations += 1;
-                } else {
-                    stop_size_bytes_exceeded_iterations = 0;
-                }
+            if is_exceeded_stop_size_iterations(
+                storage_previous,
+                stop_size_bytes,
+                &mut stop_size_bytes_exceeded_iterations,
+                stop_size_iterations,
+            ) {
+                error!(
+                    "Storage size exceeds the configured limit {} for {} times",
+                    stop_size_bytes, stop_size_shreds_exceeded_iterations,
+                );
```

Comment:
nit: What do you think of moving the `error!()` into the helper function to save duplicating the error!() print? I don't feel too strongly, so feel free to disregard this one too

```
            if is_exceeded_stop_size_iterations(
                storage_previous,
                stop_size_bytes,
                &mut stop_size_bytes_exceeded_iterations,
                stop_size_iterations,
                "Total"
            ) {
...
            if is_exceeded_stop_size_iterations(
                data_shred_storage_previous,
                stop_size_shreds_bytes,
                &mut stop_size_shreds_exceeded_iterations,
                stop_size_shreds_iterations,
                "Data shred"
            ) {
```